### PR TITLE
Screwdriver driven development.

### PR DESCRIPTION
**Pull request by draknyte1**
Added NEI handler for TGS.
Added BoP support to TGS.
Added base work for LFH.

### CODE REVIEW

**Comment by iouter** [comment]
i recommend letting vanilla trees don't need fert in TGS. (i think need nerf the Farming Station for tps)

**Comment by draknyte1** [comment]
> i recommend letting vanilla trees don't need fert in TGS. (i think need nerf the Farming Station for tps)

Nothing needs fert. It is optional if you want saplings.

**Comment by Dream-Master** [comment]
tested ingame seems to work

File: src/main/java/gtPlusPlus/core/recipe/RECIPES_Machines.java [hunk 1]
```diff
@@ -1299,6 +1310,12 @@ private static void runModRecipes(){
 					aBronzeBricks, ALLOY.TUMBAGA.getFrameBox(1), aBronzeBricks,
 					aBronzeBricks, ALLOY.TUMBAGA.getGear(1), aBronzeBricks,
 					GregtechItemList.Controller_SteamMaceratorMulti.get(1));
+			// Steam Compressor Multi
+			RecipeUtils.addShapedGregtechRecipe(
+					aBronzeBricks, aBronzeBricks, aBronzeBricks,
+					ALLOY.TUMBAGA.getGear(1), ALLOY.TUMBAGA.getFrameBox(1), ALLOY.TUMBAGA.getGear(1),
```
Comment (Dream-Master) [inline src/main/java/gtPlusPlus/core/recipe/RECIPES_Machines.java:1316]: I would suggest add here two pistons

Comment (draknyte1) [reply src/main/java/gtPlusPlus/core/recipe/RECIPES_Machines.java:1316]: Sure.

File: src/main/java/gtPlusPlus/core/recipe/RECIPES_Machines.java [hunk 2]
```diff
@@ -276,6 +277,16 @@ private static void run(){
 
 	}
 	
+
+	private static void thermicFluidHeater() {
+		
+		RecipeUtils.addShapedGregtechRecipe(
+				CI.getPlate(5, 1), CI.circuitTier5, CI.getPlate(5, 1),
+				pipeTier4, ItemList.Machine_IV_FluidHeater.get(1), pipeTier4,
```
Comment (Dream-Master) [inline src/main/java/gtPlusPlus/core/recipe/RECIPES_Machines.java]: i would using iv tiered fluid pipes here (is steel atm)

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**